File: ProjetASI2/PersonalFinanceManagementAppFrontEnd/src/app/Model/BudgetHistory.ts
import { Person } from "./Person";

export class BudgetHistory{
    id:Number;
    person:Person | null;
    totalBudget: Number;
    oldBudget:Number;
    newBudget:Number;
    dateModified:Date| String;

    constructor(id:Number, person:Person, totalBudget: Number, oldBudget:Number, newBudget:Number, dateModified:Date| String){
        this.id= id;
        this.person = person;
        this.totalBudget = totalBudget;
        this.oldBudget = oldBudget;
        this.newBudget = newBudget;
        this.dateModified = dateModified;
    }

}

File: ProjetASI2/Personal-Finance-Management-Spring-boot/src/main/java/com/PersonalFinanceManagementApplication/Personal/Finance/Management/Application/Model/BudgetHistory.java
package com.PersonalFinanceManagementApplication.Personal.Finance.Management.Application.Model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "Budget_History")
@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class BudgetHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    public BudgetHistory(Person person, BigDecimal totalBudget ,BigDecimal oldBudget, BigDecimal newBudget, LocalDateTime dateModified) {
        this.person = person;
        this.oldBudget = oldBudget;
        this.newBudget = newBudget;
        this.dateModified = dateModified;
        this.totalBudget = totalBudget;
    }

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "person_id")
    @JsonIgnore
    private Person person;

    private BigDecimal totalBudget;

    private BigDecimal oldBudget;

    private BigDecimal newBudget;


    private LocalDateTime dateModified;
}
